src/ImageSharp/Formats/Jpeg/Components/Encoder/RgbToYCbCrConverterVectorized.cs

```diff
@@ -94,27 +130,130 @@ public static void Convert(ReadOnlySpan<Rgb24> rgbSpan, ref Block8x8F yBlock, re
+                    Unsafe.Add(ref yBlockLeft.V0, yBlockVerticalOffset) = SimdUtils.HwIntrinsics.MultiplyAdd(SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(f0114, b), f0587, g), f0299, r);
+
+                    rDataLanes[j] = r;
+                    gDataLanes[j] = g;
+                    bDataLanes[j] = b;
+                }
+
+                // 16x2 => 8x1
+                // right 8x8 column conversions
+                for (int j = 1; j < 4; j += 2)
+                {
+                    rgb = Avx2.PermuteVar8x32(Unsafe.AddByteOffset(ref rgbByteSpan, (IntPtr)(bytesPerRgbStride * (i * 4 + j))).AsUInt32(), extractToLanesMask).AsByte();
+
+                    rgb = Avx2.Shuffle(rgb, extractRgbMask);
+
+                    rg = Avx2.UnpackLow(rgb, zero);
+                    bx = Avx2.UnpackHigh(rgb, zero);
+
+                    r = Avx.ConvertToVector256Single(Avx2.UnpackLow(rg, zero).AsInt32());
+                    g = Avx.ConvertToVector256Single(Avx2.UnpackHigh(rg, zero).AsInt32());
+                    b = Avx.ConvertToVector256Single(Avx2.UnpackLow(bx, zero).AsInt32());
+
+                    int yBlockVerticalOffset = (i * 2) + ((j & 2) >> 1);
+
+                    // (0.299F * r) + (0.587F * g) + (0.114F * b);
+                    Unsafe.Add(ref yBlockRight.V0, yBlockVerticalOffset) = SimdUtils.HwIntrinsics.MultiplyAdd(SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(f0114, b), f0587, g), f0299, r);
+
+                    rDataLanes[j] = r;
+                    gDataLanes[j] = g;
+                    bDataLanes[j] = b;
+                }
+
+                r = Scale_8x4_4x2(rDataLanes);
+                g = Scale_8x4_4x2(gDataLanes);
+                b = Scale_8x4_4x2(bDataLanes);
+
+                // 128F + ((-0.168736F * r) - (0.331264F * g) + (0.5F * b))
+                Unsafe.Add(ref destCbRef, i) = Avx.Add(f128, SimdUtils.HwIntrinsics.MultiplyAdd(SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(f05, b), fn0331264, g), fn0168736, r));
+
+                // 128F + ((0.5F * r) - (0.418688F * g) - (0.081312F * b))
+                Unsafe.Add(ref destCrRef, i) = Avx.Add(f128, SimdUtils.HwIntrinsics.MultiplyAdd(SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(fn0081312F, b), fn0418688, g), f05, r));
+            }
 #endif
         }
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector256<float> Scale_8x4_4x2(Span<Vector256<float>> v)
+        {
+            Vector256<int> switchInnerDoubleWords = Unsafe.As<byte, Vector256<int>>(ref MemoryMarshal.GetReference(SimdUtils.HwIntrinsics.PermuteMaskSwitchInnerDWords8x32));
+            var f025 = Vector256.Create(0.25f);
+
+            Vector256<float> topPairSum = SumHorizontalPairs(v[0], v[2]);
+            Vector256<float> botPairSum = SumHorizontalPairs(v[1], v[3]);
+
+            return Avx2.PermuteVar8x32(Avx.Multiply(SumVerticalPairs(topPairSum, botPairSum), f025), switchInnerDoubleWords);
```

Comment:
Looks like this is refactored from a previous PR I missed, but the SIMD here is a bit wacky.  If I'm reading this right, it does something like this:

```
v0 = 0 | 1 | 2 | 3 || 4 | 5 | 6 | 7
v1 = 8 | 9 | a | b || c | d | e | f
v2 = g | h | i | j || k | l | m | n
v3 = o | p | q | r || s | t | u | v

vs0 = vshufps v0,v2, mask 0,2,0,2
vs1 = vshufps v1,v3, mask 0,2,0,2
vs2 = vshufps v0,v2, mask 1,3,1,3
vs3 = vshufps v1,v3, mask 1,3,1,3

vs0 = 0 | 2 | g | i || 4 | 6 | k | m
vs1 = 8 | a | o | q || c | e | s | u
vs2 = 1 | 3 | h | j || 5 | 7 | l | n
vs3 = 9 | b | p | r || d | f | t | v

v0 = vaddps vs0,vs2
v1 = vaddps vs1,vs3

v0 = 0+1|2+3|g+h|i+j||4+5|6+7|k+l|m+n
v1 = 8+9|a+b|o+p|q+r||c+d|e+f|s+t|u+v

vs0 = vshufps v0,v1 mask 0,1,0,1
vs1 = vshufps v0,v1 mask 2,3,2,3

vs0 = 0+1|2+3|8+9|a+b||4+5|6+7|c+d|e+f
vs1 = g+h|i+j|o+p|q+r||k+l|m+n|s+t|u+v

v0 = vaddps vs0,vs1
v0 = vmulps v0, 0.25f

v0 = 01gh|23ij|89op|abqr||45kl|67mn|cdst|efuv

v0 = vpermd v0, mask 0,1,4,5,2,3,6,7

v0 = 01gh|23ij|45kl|67mn||89op|abqr|cdst|efuv
```

Which could be dramatically simplified by using horizontal adds, like such

```
v0 = 0 | 1 | 2 | 3 || 4 | 5 | 6 | 7 
v1 = 8 | 9 | a | b || c | d | e | f
v2 = g | h | i | j || k | l | m | n
v3 = o | p | q | r || s | t | u | v

v0 = vhaddps v0,v1
v1 = vhaddps v2,v3

v0 = 0+1|2+3|8+9|a+b||4+5|6+7|c+d|e+f
v1 = g+h|i+j|o+p|q+r||k+l|m+n|s+t|u+v

v0 = vaddps v0,v1
v0 = vmulps v0, 0.25f

v0 = 01gh|23ij|89op|abqr||45kl|67mn|cdst|efuv

v0 = vpermq v0, mask 0,2,1,3

v0 = 01gh|23ij|45kl|67mn||89op|abqr|cdop|efqr
```

Note that `vpermq` (Avx2.Permute4x64) works here because float pairs are kept together, and that instruction takes an immediate byte mask rather than a vector mask.

Usual disclaimer I haven't tested this, but it should be close if not completely correct.

Reply:
Completely forgot about horizontal adds, thanks a lot! Will try it today with benchmarks.